joblib/test/test_parallel.py

```diff
@@ -151,19 +159,52 @@ def test_main_thread_renamed_no_warning(backend, monkeypatch):
         results = Parallel(n_jobs=2, backend=backend)(
             delayed(square)(x) for x in range(3))
         assert results == [0, 1, 4]
+
+    # Due to the default parameters of LokyBackend, there is a change that
+    # warninfo catches Warnings from worker timeouts. We remove it if it exists
+    warninfo = [w for w in warninfo if "worker timeout" not in str(w.message)]
+
     # The multiprocessing backend will raise a warning when detecting that is
     # started from the non-main thread. Let's check that there is no false
     # positive because of the name change.
     assert len(warninfo) == 0
 
 
+# TODO: The Loky backend can be nested. Do we want that?
+@with_multiprocessing
+@parametrize('child_backend', PROCESS_BACKENDS)
+@parametrize('parent_backend', ['multiprocessing', 'threading'])
+def test_nested_parallel_warnings(parent_backend, child_backend, capfd):
+    if posix is None:
+        # This test pass only when fork is the process start method
+        raise SkipTest('Not a POSIX platform')
+
+    # Makes sure the warnings are always printed
+    warnings.resetwarnings()
+    warnings.simplefilter("always")
+
+    # no warnings if inner_n_jobs=1
```

Comment:
Ok for the context manager.

Regarding the parametrize, it can still be done but you'll need to adapt the assert with the inner_n_jobs value (that's what I would do).